tools/cli-core/src/util.ts

```diff
@@ -170,3 +177,44 @@ export const generateJSONFile = ( filePath: string, data: unknown ) => {
 	);
 	return writeFile( filePath, json );
 };
+
+/**
+ * Get plugin data from main PHP file in JSON format.
+ *
+ * @param {string} tmpRepoPath    - Path to repo.
+ * @param {string} pathToMainFile - Path to plugin's main PHP file.
+ * @param {string} hashOrBranch   - Hash or branch to checkout.
+ * @return {Object} - plugin data in JSON format.
+ */
+export const getPluginData = async (
+	tmpRepoPath: string,
+	pathToMainFile: string,
+	hashOrBranch: string
+) => {
+	const git = simpleGit( { baseDir: tmpRepoPath } );
+	await git.checkout( [ hashOrBranch ] );
+
+	const mainFile = join( tmpRepoPath, pathToMainFile );
+
+	const content = await readFile( mainFile, 'utf-8' );
+	const regex = /<\?php\n\/\*\*\n([\S\s]+?) \*\//;
+	const match = content.match( regex );
+
+	if ( ! match || match.length < 2 ) {
+		Logger.error( 'Failed to find plugin data!' );
+		return;
+	}
+
+	return match[ 1 ]
+		.replace( /@package .+\n/, '' )
+		.split( '*' )
+		.reduce( ( result: { [ key: string ]: string }, line: string ) => {
```

Comment:
the type `{ [ key: string ]: string }` can be more simply and easily read typed as `Record<string, string>`. nit pick but just letting you know :)

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/34951/commits/b677f302a202dc1e1bbb93414481f6aec5fb10c7